### PR TITLE
feature(main): sealos run not save clusterfile when init failed

### PR DESCRIPTION
**Pull request by cuisongliu**
[SKIP CI]sealos: 一句话简短描述该 PR 内容

### CODE REVIEW

**Review by berlinsaint — APPROVED** [review]

**Review by berlinsaint — APPROVED** [review]
/approved

**Review by berlinsaint — APPROVED** [review]

**Review by berlinsaint — APPROVED** [review]

File: pkg/types/v1beta1/cluster.go [hunk 1]
```diff
@@ -61,8 +62,30 @@ type MountImage struct {
...
+
+type ClusterPhase string
+
+const (
+	ClusterFailed    ClusterPhase = "ClusterFailed"
+	ClusterSuccess   ClusterPhase = "ClusterSuccess"
+	ClusterInProcess ClusterPhase = "ClusterInProcess"
+)
+
+// ClusterCondition describes the state of a aijob at a certain point.
```
Comment (SignorMercurio) [inline pkg/types/v1beta1/cluster.go:74]: typo? `aijob` -> `job`

**Review by SignorMercurio — APPROVED** [review]

File: pkg/apply/applydrivers/apply_drivers_default.go [hunk 2]
```diff
@@ -67,19 +69,43 @@ type Applier struct {
...
 	logger.Debug("write cluster file to local storage: %s", clusterPath)
 	return yaml.MarshalYamlToFile(clusterPath, c.ClusterDesired)
 }
+func (c *Applier) initStatus() {
+	c.ClusterDesired.Status.Phase = v2.ClusterInProcess
+	c.ClusterDesired.Status.Conditions = make([]v2.ClusterCondition, 0)
+}
+
+func (c *Applier) finalStatus(err error) {
+	logger.Error("final status error: %v", err)
```
Comment (fanux) [inline pkg/apply/applydrivers/apply_drivers_default.go:90]: what dose final status error means.

File: pkg/apply/applydrivers/apply_drivers_default.go [hunk 3]
```diff
@@ -67,19 +69,43 @@ type Applier struct {
...
+	condition := v2.ClusterCondition{
+		Type:              "ApplierApplySuccess",
+		Status:            v1.ConditionTrue,
+		LastHeartbeatTime: metav1.Now(),
+		Reason:            "Ready",
+		Message:           "applier apply successfully",
```
Comment (fanux) [inline pkg/apply/applydrivers/apply_drivers_default.go:96]: apply cluster successfully

Comment (SignorMercurio) [reply pkg/apply/applydrivers/apply_drivers_default.go:96]: Applied successfully. / Applied to cluster successfully.

File: pkg/apply/processor/create.go [hunk 4]
```diff
@@ -77,6 +78,7 @@ func (c *CreateProcessor) GetPipeLine() ([]func(cluster *v2.Cluster) error, erro
 	return todoList, nil
 }
 func (c *CreateProcessor) Check(cluster *v2.Cluster) error {
+	logger.Info("create process exec pipeline: Check")
```
Comment (SignorMercurio) [inline pkg/apply/processor/create.go:81]: Error executing pipeline Check in CreateProcessor

File: pkg/apply/processor/create.go [hunk 5]
```diff
@@ -103,6 +105,7 @@ func (c *CreateProcessor) CheckImageType(cluster *v2.Cluster) error {
 	return nil
 }
 func (c *CreateProcessor) PreProcess(cluster *v2.Cluster) error {
+	logger.Info("create process exec pipeline: PreProcess")
```
Comment (SignorMercurio) [inline pkg/apply/processor/create.go:108]: Executing pipeline PreProcess in CreateProcessor.

File: pkg/apply/processor/create.go [hunk 6]
```diff
@@ -149,6 +153,7 @@ func (c *CreateProcessor) RunConfig(cluster *v2.Cluster) error {
 }
 
 func (c *CreateProcessor) MountRootfs(cluster *v2.Cluster) error {
+	logger.Info("create process exec pipeline: MountRootfs")
```
Comment (SignorMercurio) [inline pkg/apply/processor/create.go:156]: Executing pipeline MountRootfs in CreateProcessor.

File: pkg/apply/processor/create.go [hunk 7]
```diff
@@ -158,10 +163,12 @@ func (c *CreateProcessor) MountRootfs(cluster *v2.Cluster) error {
 }
 
 func (c *CreateProcessor) Init(cluster *v2.Cluster) error {
+	logger.Info("create process exec pipeline: Init")
```
Comment (SignorMercurio) [inline pkg/apply/processor/create.go:166]: Executing pipeline Init in CreateProcessor.

File: pkg/apply/processor/create.go [hunk 8]
```diff
@@ -178,6 +185,7 @@ func (c *CreateProcessor) Join(cluster *v2.Cluster) error {
 }
 
 func (c *CreateProcessor) RunGuest(cluster *v2.Cluster) error {
+	logger.Info("create process exec pipeline: RunGuest")
```
Comment (SignorMercurio) [inline pkg/apply/processor/create.go:188]: Executing pipeline RunGuest in CreateProcessor.

File: pkg/apply/processor/scale.go [hunk 9]
```diff
@@ -85,13 +85,15 @@ func (c *ScaleProcessor) GetPipeLine() ([]func(cluster *v2.Cluster) error, error
 	return todoList, nil
 }
 func (c *ScaleProcessor) Delete(cluster *v2.Cluster) error {
+	logger.Info("scale process exec pipeline: Delete")
```
Comment (SignorMercurio) [inline pkg/apply/processor/scale.go:88]: Executing pipeline Delete in ScaleProcessor.

File: pkg/apply/processor/scale.go [hunk 10]
```diff
@@ -85,13 +85,15 @@ func (c *ScaleProcessor) GetPipeLine() ([]func(cluster *v2.Cluster) error, error
...
 	err := c.Runtime.DeleteMasters(c.MastersToDelete)
 	if err != nil {
 		return err
 	}
 	return c.Runtime.DeleteNodes(c.NodesToDelete)
 }
 func (c *ScaleProcessor) Join(cluster *v2.Cluster) error {
+	logger.Info("scale process exec pipeline: Join")
```
Comment (SignorMercurio) [inline pkg/apply/processor/scale.go:96]: Executing pipeline Join in ScaleProcessor.

File: pkg/apply/processor/scale.go [hunk 11]
```diff
@@ -129,6 +133,7 @@ func (c *ScaleProcessor) JoinCheck(cluster *v2.Cluster) error {
 }
 
 func (c *ScaleProcessor) DeleteCheck(cluster *v2.Cluster) error {
+	logger.Info("scale process exec pipeline: DeleteCheck")
```
Comment (SignorMercurio) [inline pkg/apply/processor/scale.go:136]: Executing pipeline DeleteCheck in ScaleProcessor.

File: pkg/apply/processor/scale.go [hunk 12]
```diff
@@ -141,6 +146,7 @@ func (c *ScaleProcessor) DeleteCheck(cluster *v2.Cluster) error {
 }
 
 func (c *ScaleProcessor) PreProcess(cluster *v2.Cluster) error {
+	logger.Info("scale process exec pipeline: PreProcess")
```
Comment (SignorMercurio) [inline pkg/apply/processor/scale.go:149]: Executing pipeline Preprocess in ScaleProcessor.

File: pkg/apply/processor/scale.go [hunk 13]
```diff
@@ -164,6 +170,7 @@ func (c *ScaleProcessor) PreProcess(cluster *v2.Cluster) error {
 }
 
 func (c *ScaleProcessor) RunConfig(cluster *v2.Cluster) error {
+	logger.Info("scale process exec pipeline: RunConfig")
```
Comment (SignorMercurio) [inline pkg/apply/processor/scale.go:173]: Executing pipeline RunConfig in ScaleProcessor.

**Review by fanux — APPROVED** [review]

**Pull request merged**